aiohttp/wsgi.py

```
@@ -159,20 +159,39 @@ def handle_request(self, message, payload):
         response = self.create_wsgi_response(message)
 
         riter = self.wsgi(environ, response.start_response)
-        if isinstance(riter, asyncio.Future) or inspect.isgenerator(riter):
+        if isinstance(riter, asyncio.Future):
             riter = yield from riter
+        elif inspect.isgenerator(riter):
+            while True:
+                try:
+                    item = next(riter)
+                except StopIteration as e:
+                    riter = e.value
+                    break
+                if isinstance(item, asyncio.Future):
+                    try:
+                        yield item
+                    except:
```

application wont be able to get exception from future.
for example if application waits on socket operation this code swallow all socket exceptions.